client/packet_test.go

```diff
@@ -0,0 +1,144 @@
+package client
+
+import (
+	"net"
+	"reflect"
+	"testing"
+)
+
+var mac = net.HardwareAddr([]byte{0xb8, 0xae, 0xed, 0x7a, 0x10, 0x66})
+
+func TestNewSolicitOptions(t *testing.T) {
+	options, err := newSolicitOptions(mac)
+	if err != nil {
+		t.Fatalf("error in newSolicitOptions: %v\n", err)
+	}
+	expected := Options(map[OptionCode][][]byte{
+		OptionIANA:        [][]byte{[]byte{0x72, 0x6f, 0x6f, 0x74, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
+		OptionRapidCommit: [][]byte{nil},
+		OptionElapsedTime: [][]byte{[]byte{0x00, 0x00}},
+		OptionORO:         [][]byte{[]byte{0x00, byte(OptionDNSServers), 0x00, byte(OptionDomainList), 0x00, byte(OptionBootFileURL), 0x00, byte(OptionBootFileParam)}},
+		OptionClientID:    [][]byte{[]byte{0x00, 0x03, 0x00, 0x06, 0xb8, 0xae, 0xed, 0x7a, 0x10, 0x66}},
+	})
+
+	optionsIANA, optionsSuccess, err := options.IANA()
+	if err != nil {
+		t.Fatalf("getting IANA from options: got %v, want nil", err)
+	}
+	if optionsSuccess != true {
+		t.Fatalf("incorrect newSolicitOptions: IANA does not exist\n")
+	}
+
+	expectedIANA, expectedSuccess, err := expected.IANA()
+	if err != nil {
+		t.Fatalf("getting IANA from expected: got %v, want nil", err)
+	}
+	if expectedSuccess != true {
+		t.Fatalf("incorrect newSolicitOptions: IANA does not exist\n")
+	}
+	if !reflect.DeepEqual(optionsIANA, expectedIANA) {
```

Comment:
I'm a big fan of https://godoc.org/github.com/google/go-cmp/cmp#Diff nowadays for these types of comparisons, as it provides a really easy to read diff view of what is wrong.